File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = INT_MIN;
        int buyPrice = prices[0];
        for(int i=1;i<prices.size();i++){
            if(prices[i]<buyPrice)
            buyPrice = prices[i];
            else{
                profit = max(profit,prices[i]-buyPrice);
            }
        }
        if(profit==INT_MIN)
        return 0;
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
import java.util.*;
class Solution {
    public int maxProfit(int[] prices) {
        int minVal = prices[0];int profit = 0;
        for(int i=1;i<prices.length;i++){
            if(prices[i]>=minVal)
            {
                profit=Math.max(profit,prices[i]-minVal); 
            }
            else{
                minVal = prices[i];
            }
        }
        return profit;
    }
}